packages/sanity/src/_internal/cli/actions/schema/schemaListAction.ts

```diff
@@ -0,0 +1,100 @@
+import {type CliCommandArguments, type CliCommandContext, type CliOutputter} from '@sanity/cli'
+import {type SanityDocument} from '@sanity/client'
+import chalk from 'chalk'
+import {size, sortBy} from 'lodash'
+
+export interface SchemaListFlags {
+  json: boolean
+  id: string
+}
+
+type PrintSchemaListArgs = {
+  schemas: SanityDocument[]
+  output: CliOutputter
+  dataset: string
+  projectId: string
+}
+
+export const SANITY_WORKSPACE_SCHEMA_ID = 'sanity.workspace.schema'
+
+const printSchemaList = ({schemas, output, dataset, projectId}: PrintSchemaListArgs) => {
+  const ordered = sortBy(
+    schemas.map(({_createdAt: createdAt, _id: id, workspace}) => {
+      return [id, workspace.title, dataset, projectId, createdAt].map(String)
+    }),
+    ['createdAt'],
+  )
+  const headings = ['Id', 'Title', 'Dataset', 'ProjectId', 'CreatedAt']
+  const rows = ordered.reverse()
+
+  const maxWidths = rows.reduce(
+    (max, row) => row.map((current, index) => Math.max(size(current), max[index])),
+    headings.map((str) => size(str)),
+  )
+
+  const printRow = (row: string[]) => row.map((col, i) => `${col}`.padEnd(maxWidths[i])).join('   ')
+
+  output.print(chalk.cyan(printRow(headings)))
+  rows.forEach((row) => output.print(printRow(row)))
+}
+
+export default async function storeSchemaAction(
```

Comment:
Export default because lazyloaded /established pattern for actions in the CLI right?